helpers.lua

```diff
@@ -134,6 +136,61 @@ end
 
 -- }}}
 
+
+-- {{{ Network functions
+
+-- Create a recieving buffer
+-- NOTE: Not sure whether to export this with helpers or not
+-- Probably just gonna leave it local for now
+function generate_buffer(buffer_length)
+    -- First, create an output buffer to recieve from
+    local recv_buffer = " "
+    -- We'll need to allocate `buffer_length` bytes to it
+    for i=1,buffer_length do
+        recv_buffer = recv_buffer .. " "
+    end
+
+    return recv_buffer
+end
```

Comment:
`return string.rep(" ", buffer_length + 1)` (the `+1` is there to match the behaviour of your code, but I don't know if your code really should do that)